improve formatting and add a bit to the `Goals and..` section

zetok opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zetok/tox-spec/2)
<!-- Reviewable:end -->

zetok committed:
improve formatting and add a bit to the `Goals and..` section

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
+(TODO: this is just a placeholder; some more technical description should be
+given)
 
 This section should give an idea on what are the goals and non-goals of Tox, so
```
This section should give an idea on what the goals and non-goals of Tox are, so that the reader:

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
+#### End-to-end encryption
+
+All the messages are encrypted via keys derived via DH, thus keys are only
+known to sender and receiver and are never transferred over network
```
All the messages are encrypted with keys derived via DH, thus keys are only known to sender and receiver and are never transferred over the network.

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
 
-(TODO: this is just a placeholder; some more technical description should be
-given)
+### What Tox Does
```
Heading capitalisation is inconsistent

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
+#### Anonymity
 
-What Tox Does NOT:
+Unless TCP mode is used, participants communicate directly to each other. One
```
with each other

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
 
-What Tox Does NOT:
+Unless TCP mode is used, participants communicate directly to each other. One
+of the reasons for this is that relaying real-time video is rather too costly
```
rather costly OR too costly

robinlinden replied:
relaying real-time video with anything but a direct connection

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
-What Tox Does NOT:
+Unless TCP mode is used, participants communicate directly to each other. One
+of the reasons for this is that relaying real-time video is rather too costly
+(in terms of load) and also means delays.
```
Maybe (in terms of bandwidth)?

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
-    communicate directly to each other; One of the reasons for this is that
-    relaying real-time video is rather too costly (in terms of load) and also
-    means delays;
+Your IP Address is exposed to nodes in your friendlist. They can link your ID
```
IP address

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
-    relaying real-time video is rather too costly (in terms of load) and also
-    means delays;
+Your IP Address is exposed to nodes in your friendlist. They can link your ID
+directly to IP Address.
```
to your IP address

robinlinden commented on spec.md:
```diff
@@ -22,67 +22,78 @@ bytes (e.g. `32`), a number in bits (e.g. `7` bit), a choice of lengths (e.g.
-    -   Your IP Address is exposed to nodes in your friendlist; They can link
-        your ID directly to IP Address;
+Temporary DHT nodes and onion tunnels are used to find friends, so that your ID
+cannot be linked to your IP based solely on publicly available data (TODO: i.e.
```
Replace i.e. with e.g. 
I.e. is for putting things in other words. E.g. is for examples.

zetok commented:
Done.

robinlinden reviewed (approved):
LGTM.

zetok committed:
get sufficiently new pandoc with formatting bug fixed

zetok committed:
correct formatting to conform with newest pandoc

Fix to https://github.com/jgm/pandoc/issues/3277 causes slight
formatting changes.

zetok committed:
speed up travis build by caching stack & build pandoc only when needed
